backend/src/api/services/question/question.service.ts

```diff
@@ -45,14 +45,34 @@ export const getSidebarMetadata = async (studentID: string) => {
     return sidebarMetada;
 };
 
-export const updateUserBookmark = async (studentID: string, questionID: string) => {
-    const students: [StudentProgress] = await loadData('progress.json');
-    const student = students.find(student => student.studentID === studentID);
+/*
+If student wants to unbookmark, set "bookmark" parameter to false.
+*/
+export const updateUserBookmark = async (studentID: string, questionID: string, bookmark: boolean=true) => {
+    // Check if the question exists
+    const question = await QuestionBaseModel.findById(questionID);
+    if (!question) {
+        throw new HttpError(status.NOT_FOUND, `question is not found with question ID ${questionID}` );
+    }
+
+    // Find the student
+    const student = await StudentProgressModel.findOne({ studentID });
     if (!student) {
-        throw new Error('Student not found');
+        throw new HttpError(status.NOT_FOUND, `student is not found with student ID ${studentID}` );
     }
-    student.bookMark.push(questionID);
 
+    // Update user bookmark
+    if (bookmark) {
+        // Add to bookmarks
+        if (!student.bookMark.includes(questionID)) {
+            student.bookMark.push(questionID);
+        }
+    } else {
+        // Remove from bookmarks
+        student.bookMark = student.bookMark.filter(id => id !== questionID);
+    }
```

Comment:
the else branch will never be executed  because default value `bookmark` is always true unless a dynamic bookmark value is provided

Reply:
Ok let me mitigate this.